service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/GetAzureStorageContainerStep.java

```diff
@@ -0,0 +1,80 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.storageContainer;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.exception.DuplicateResourceException;
+import bio.terra.workspace.service.resource.exception.ResourceNotFoundException;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.storage.StorageManager;
+import com.azure.resourcemanager.storage.fluent.models.ListContainerItemInner;
+
+
+/**
+ * Gets an Azure Storage Container, and fails if it already exists or if the parent storage account does not exist.
+ * This step is designed to run immediately before {@link CreateAzureStorageContainerStep} to ensure idempotency
+ * of the create operation.
+ */
+public class GetAzureStorageContainerStep implements Step {
+
+  private final AzureConfiguration azureConfig;
+  private final CrlService crlService;
+  private final ControlledAzureStorageContainerResource resource;
+
+  public GetAzureStorageContainerStep(
+      AzureConfiguration azureConfig,
+      CrlService crlService,
+      ControlledAzureStorageContainerResource resource) {
+    this.azureConfig = azureConfig;
+    this.crlService = crlService;
+    this.resource = resource;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    final AzureCloudContext azureCloudContext =
+            context.getWorkingMap().get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+    final StorageManager storageManager = crlService.getStorageManager(azureCloudContext, azureConfig);
+
+    try {
+      storageManager
+              .storageAccounts()
+              .getByResourceGroup(
+                      azureCloudContext.getAzureResourceGroupId(), resource.getStorageAccountName());
+    } catch (ManagementException ex) {
+      return new StepResult(
```

Comment:
Same question about `ManagementException`, does this always mean not found?

Reply:
You are right-- I shouldn't assume it doesn't exist. However, if we can't retrieve the storage account I think we should assume that creating the container will fail. I will change the error message to be more general.

Reply:
I have improved the exception logging to be more specific.